### PR TITLE
Add SF30/SN30 Pro controller firmware 1.23 (Closes #1)

### LINKED ISSUE #1: New devices and firmwares
New 8bitdo devices and firmwares have been released:
http://support.8bitdo.com/#tab2

Are pull requests welcome to add support for those ?

### PR DESCRIPTION
**Pull request by superm1**

### CODE REVIEW

**Comment by superm1** [comment]
So the part I'm unclear on is if it's really a good idea to match to all the VID/PID pairs or just one of them. The reason why detailed in https://github.com/hughsie/fwupd/pull/354

File: releases/SF30_Pro-SN30_Pro/1.23/sf30sn30pro.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  <developer_name>8Bitdo</developer_name>
+  <releases>
+    <release urgency="medium" version="1.23" date="2017-12-28">
+      <checksum filename="Firmware_SF30&amp;SN30_Pro_V1.23.dat" target="content"/>
+      <description>
+        <p>
+          This stable release fixes the following issues:
+        </p>
+        <ul>
+          <li>1.Enhanced vibration optimization on Xinput mode.</li>
```
Comment (hughsie) [inline releases/SF30_Pro-SN30_Pro/1.23/sf30sn30pro.metainfo.xml]: No need for `1.` (it's an unordered list..) and no need for the trailing dot.

File: releases/SF30_Pro-SN30_Pro/1.23/sf30sn30pro.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          <li>2.Added USB connection to Mac mode.</li>
+          <li>3.Fixed the bug that the controller would not turn on occasionally. Enhance the stability of the Bluetooth automatic connection</li>
```
Comment (hughsie) [inline releases/SF30_Pro-SN30_Pro/1.23/sf30sn30pro.metainfo.xml]: Isn't this two `li`'s?

**Comment by hughsie** [comment]
If I understand correctly, it's the VID for Nintendo, so I don't think we should match on that specific one... I'll ask our friendly designer to do some line art for the screenshot. Thanks so far!

File: releases/SF30_Pro-SN30_Pro/1.23/sf30sn30pro.metainfo.xml (hunk 1 above)
Comment (superm1) [reply releases/SF30_Pro-SN30_Pro/1.23/sf30sn30pro.metainfo.xml]: OK

File: releases/SF30_Pro-SN30_Pro/1.23/sf30sn30pro.metainfo.xml (hunk 2 above)
Comment (superm1) [reply releases/SF30_Pro-SN30_Pro/1.23/sf30sn30pro.metainfo.xml]: Yep, will adjust.

**Comment by superm1** [comment]
Well so I "think" fwupd should ideally show the controller no matter what mode it's in (Eg the Nintendo or Sony VID come up too).  Otherwise that means you have to know to put the controller in Dinput mode for it to show up in fwupd and then in bootloader mode after that to flash it with fwupd.  That's why I think it makes sense to match all of the VID/PID pairs it can behave as.

**Comment by hughsie** [comment]
> That's why I think it makes sense to match all of the VID/PID pairs it can behave as

I agree it should match, but I also think it shouldn't be `UPGRADABLE` so to speak. Can we maybe use something other than `none` as the quirk and set the appropriate flag based on that? The description for the device could even mention that it needs to be a different mode.

**Comment by superm1** [comment]
Updated with new artwork.

**Pull request merged**